### ISSUE TITLE
Support for nullable fields

### ISSUE THREAD

**Issue by dinkelspiel** [giacomocavalieri/squirrel#51]
Currently when I have a row that does not specify `NOT NULL` i'd expect the parameter in the resulting function to take an Option but I currently see no way of doing that. To get around this I've created two sql files but this feels like something that should be provided.

**Comment by giacomocavalieri** [comment]
Yes! I really want a way to support this, sadly Postgres doesn't give us enough information to get this kind of info and it will probably have to be implemented as some sort of annotation.
Did you have any design in mind for how that could look?

**Comment by dinkelspiel** [comment]
I was mostly hoping for each nullable field would be a `Option(a)` instead of just `a`. This would be breaking however

**Comment by dinkelspiel** [comment]
I was mostly hoping for each nullable field would be a Option(a) instead of just a. This would be breaking however.
For example given the pseudo sql below:
```sql
username NOT NULL
email NOT NULL
profile_picture NULLABLE
```
then the generated squirrel code would go from (like it is right now)
```rs
sql.create_user(username: String, email: String, profile_picture: String)
```
to
```rs
sql.create_user(username: String, email: String, profile_picture: Option(String))
```

**Comment by giacomocavalieri** [comment]
Yeah I get that but Postgres won't give us the kind of information we need to figure out if some query parameter is nullable or not. So it would most likely need to be implemented as some sort of extension with a magic comment or something like that

**Comment by dinkelspiel** [comment]
Gotchu, that is problematic. I think keeping away from magic comments are some best things with gleam so i think that should be avoided. If you were to go that way however then you could do something like [frontmatter](https://frontmatter.codes/docs/markdown) where the first comment in a document is yaml or some config language* that is read and then parsed as a file config so you'd do.

```sql
/*
nullable:
  - profile_picture
*/

INSERT INTO user(username, profile_picture) VALUES($1, $2)
```

Where this would make `profile_picture` or `$2` an `Option(String)` but that would open up to runtime errors since you could accidentally break a not null constraint.

**Comment by giacomocavalieri** [comment]
I'm closing this in favour of #23, we can further discuss how to tackle this problem in that discussion thread
